client-java/src/unit-tests/java/uk/gov/gchq/palisade/client/ClientManagerTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.client;
+
+import org.junit.jupiter.api.Test;
+
+import uk.gov.gchq.palisade.client.internal.dft.DefaultClient;
+import uk.gov.gchq.palisade.client.internal.dft.DefaultSession;
+
+import java.net.URI;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ClientManagerTest {
+
+    @Test
+    void testGetClient() {
+        var client = ClientManager.getClient("pal://alice@localhost:1234/cluster");
+        assertThat(client).isInstanceOf(DefaultClient.class);
+        client = ClientManager.getClient("pal:dft://alice@localhost:1234/cluster");
+        assertThat(client).isInstanceOf(DefaultClient.class);
+    }
+
+    @Test
+    void testOpenSessionUrlNoProperties() {
+
+        var serviceUrl = "pal://alice@localhost:1234/cluster";
```

Comment:
Suggested removing this uri encoding

Reply:
No encoding going on here.

Reply:
Apologies, rushed comment.
Suggest removing the userinfo from the uri as it should not be a supported way of configuring the client.

Reply:
What if it is needed for security to get through a load balancer?

Reply:
If it's supplied, it will simply be passed through. If it's not supported, it will be ignored by the server.